src/uagents/mailbox.py

```diff
@@ -14,32 +16,82 @@
 
 
 class MailboxClient:
-    def __init__(
-        self, agent, config: Dict[str, str], logger: Optional[logging.Logger] = None
-    ):
-        self._base_url = config["base_url"]
-        self._api_key = config["api_key"]
+    def __init__(self, agent, logger: Optional[logging.Logger] = None):
         self._agent = agent
         self._access_token: str = None
         self._poll_interval = MAILBOX_POLL_INTERVAL_SECONDS
         self._logger = logger or get_logger("mailbox")
 
+    @property
+    def base_url(self):
+        return self._agent.mailbox["base_url"]
+
+    @property
+    def api_key(self):
+        return self._agent.mailbox["api_key"]
+
+    @property
+    def protocol(self):
+        return self._agent.mailbox["protocol"]
+
+    @property
+    def http_prefix(self):
+        return self._agent.mailbox["http_prefix"]
+
     async def run(self):
-        self._logger.info(f"Connecting to mailbox server at {self._base_url}")
+        self._logger.info(f"Connecting to mailbox server at {self.base_url}")
         while True:
             try:
                 if self._access_token is None:
                     await self._get_access_token()
-                await self._poll_server()
+                if self.protocol == "ws":
+                    await self._open_websocket_connection()
+                else:
+                    await self._poll_server()
+                    await asyncio.sleep(self._poll_interval)
             except ClientConnectorError:
                 self._logger.exception("Failed to connect to mailbox server")
-            await asyncio.sleep(self._poll_interval)
+
+    async def _handle_envelope(self, payload: dict, session: aiohttp.ClientSession):
+        try:
+            env = Envelope.parse_obj(payload["envelope"])
+        except pydantic.ValidationError:
+            self._logger.warning("Received invalid envelope")
+            return
+
+        do_verify = not is_user_address(env.sender)
+
+        if do_verify and env.verify() is False:
+            self._logger.warning("Received envelope that failed verification")
+            return
+
+        if not dispatcher.contains(env.target):
+            self._logger.warning("Received envelope for unrecognized address")
+            return
+
+        await dispatcher.dispatch(
+            env.sender,
+            env.target,
+            env.protocol,
+            env.decode_payload(),
+        )
+
+        # delete envelope on server
```

Comment:
We could consider putting these in a deletion queue instead to increase reception throughput.

Reply:
Done.